assets/js/leaves.js

```diff
@@ -0,0 +1,566 @@
+// logic to generate a path for drawing leaves,
+// and for managing leaf details and tag behavior in the timetree
+
+import { select, selectAll } from "d3-selection";
+import { line, curveNatural } from "d3-shape";
+
+// combine into d3 object for convenience
+const d3 = {
```

Comment:
I *think* that's already the case, but I would limit d3 DOM operations to SVG nodes in the visualization.